giraffe/src/utils/fromFlux.ts

```diff
@@ -148,10 +194,35 @@ export const fromFlux = (fluxCSV: string): FromFluxResult => {
               resultColumnNames.add(tableData[i][columnName])
             }
           }
-          columns[columnKey].data[tableLength + i] = parseValue(
-            tableData[i][columnName] || columnDefault,
-            columnType
-          )
+          const value = tableData[i][columnName] || columnDefault
+          let result = null
+
+          if (value === undefined) {
+            result = undefined
+          } else if (value === 'null') {
+            result = null
+          } else if (value === 'NaN') {
+            result = NaN
+          } else if (columnType === 'boolean' && value === 'true') {
+            result = true
+          } else if (columnType === 'boolean' && value === 'false') {
+            result = false
+          } else if (columnType === 'string') {
+            result = value
+          } else if (columnType === 'time') {
+            result = Date.parse(value.trim())
+          } else if (columnType === 'number') {
+            if (value === '') {
+              result = null
+            } else {
+              const parsedValue = Number(value)
+              result = parsedValue === parsedValue ? parsedValue : value
+            }
+          } else {
+            result = null
+          }
```

Comment:
It looks like this `else` block is not necessary, since `result` is defaulted to `null` on line 198.